druntime/src/core/demangle.d

```diff
@@ -3273,3 +3273,13 @@ private struct BufSlice
     auto getSlice() inout nothrow scope { return buf[from .. to]; }
     size_t length() const scope { return to - from; }
 }
+
+extern(C) export int __d_demangle(const(char*) mangled, char* buffer, size_t bufferLength)
+{
+    import core.stdc.string: strlen;
+    if (mangled == null) return 0;
+
+    auto ret = demangle( cast(char[]) mangled[0 .. strlen(mangled)], buffer[0 .. bufferLength] );
+    buffer[ret.length] = 0;
```

Comment:
This may write past the end of `buffer`.

Also, if buffer is not large enough, `demangle` will return a GC-allocated slice, which this C API doesn't account for.